lib/sandbox.js

```diff
@@ -297,7 +298,12 @@ Sandbox.prototype.createTokenRaw = function (claims, options, cb) {
     var request = Superagent
         .post(this.url + '/api/tokens/issue')
         .set('Authorization', 'Bearer ' + this.token)
-        .send(claims);
+
+    if (this.csrfToken) {
+        request.set('X-CSRFToken', this.csrfToken)
+    }
```

Comment:
One way to DRY this up a bit would be to have a headers object defined above in a single place, something like: 

```
this.headers = { 'Authorization': 'Bearer ' + this.token};
 if (this.csrfToken) {
      headers['X-CSRFToken'] = this.csrfToken;
 }
```

and then reuse that everywhere below ([since it looks like Superagent can take an object for setting headers](https://visionmedia.github.io/superagent/#setting-header-fields)):

```
Superagent.post().set(this.headers);
```

Reply:
Thanks Chris, I cleanup this up significantly and moved the `request.set(headers)` call to the `issueRequest` function.